libres/lib/res_util/block_fs.cpp

```diff
@@ -1229,39 +1088,25 @@ void block_fs_fsync(block_fs_type *block_fs) {
 static void block_fs_fwrite__(block_fs_type *block_fs, const char *filename,
                               file_node_type *node, const void *ptr,
                               int data_size) {
+    block_fs_fseek(block_fs, node->node_offset);
+    node->status = NODE_IN_USE;
+    node->data_size = data_size;
+    file_node_set_data_offset(node, filename);
 
-#ifdef ENABLE_CACHE
-    if ((node->cache_size == data_size) &&
-        (memcmp(ptr, node->cache, data_size) == 0))
-        /* The current cache is identical to the data we are attempting to write - can leave immediately. */
-        return;
-#else
-    if (false)
-        return;
-#endif
+    /* This marks the node section in the datafile as write in progress with: NODE_WRITE_ACTIVE_START ... NODE_WRITE_ACTIVE_END */
+    file_node_init_fwrite(node, block_fs->data_stream);
 
-    else {
-        block_fs_fseek(block_fs, node->node_offset);
-        node->status = NODE_IN_USE;
-        node->data_size = data_size;
-        file_node_set_data_offset(node, filename);
+    /* Writes the actual data content. */
+    block_fs_fseek_node_data(block_fs, node);
+    util_fwrite(ptr, 1, data_size, block_fs->data_stream, __func__);
 
-        /* This marks the node section in the datafile as write in progress with: NODE_WRITE_ACTIVE_START ... NODE_WRITE_ACTIVE_END */
-        file_node_init_fwrite(node, block_fs->data_stream);
-
-        /* Writes the actual data content. */
-        block_fs_fseek_node_data(block_fs, node);
-        util_fwrite(ptr, 1, data_size, block_fs->data_stream, __func__);
+    /* Writes the file node header data, including the NODE_END_TAG. */
+    file_node_fwrite(node, filename, block_fs->data_stream);
 
-        /* Writes the file node header data, including the NODE_END_TAG. */
-        file_node_fwrite(node, filename, block_fs->data_stream);
-
-        block_fs_update_cache_node(block_fs, node, data_size, ptr);
-        block_fs->write_count++;
-        if (block_fs->fsync_interval &&
-            ((block_fs->write_count % block_fs->fsync_interval) == 0))
-            block_fs_fsync(block_fs);
-    }
+    block_fs->write_count++;
+    if (block_fs->fsync_interval &&
```

Comment:
Maybe just a curiosity, but this condition is never met in the c testsuite. If that is easy to remedy, it would be very nice :)